### ISSUE TITLE
Consolidate error handling

### ISSUE THREAD

**Issue by clifgriffin** [woocommerce/woocommerce-gateway-stripe#520]
Related to: #446 

**TLDR: We need a consistent interface / event model for intercepting and handling all errors.**

Currently there are two ways¹ that errors are generated and handled:

1. Stripe errors (issues with card numbers, etc)
2. Validation errors.²

Both have their own independent ways of handling and displaying errors. They don't use the same methodology.

Stripe errors generate an event that can be attached to (guessing that comes from Stripe.js itself, not the gateway) while validation errors do not. 

Our use case is Checkout for WooCommerce (https://www.checkoutwc.com). Our template is totally customized, so we can't rely on the Stripe gateways' error message placement. They don't make sense in our markup. The way we currently handle this is using a mutation observer. Which we hate. But it's the only way to consistently detect and display errors. 

I think the best solution would be a consistent event like `onStripeGatewayError` that passes through a consistent object. This would allow us to intercept and handle error messages in a consistent way. 

¹ By my count.
² It's not 100% clear to me why the Stripe gateway needs to be doing its own validation of checkout, but I'll assume I don't have all of the info and that it's totally necessary.

**Comment by roykho** [comment]
Thanks for the additional info but its actual exactly the same as https://github.com/woocommerce/woocommerce-gateway-stripe/issues/446 so lets follow that one and now with this added reference.
